sympy/tensor/tensor.py

```diff
@@ -2598,12 +2598,16 @@ def _eval_rewrite_as_Indexed(self, *args):
 
     def _eval_partial_derivative(self, s):
         # Evaluation like Add
-        # TODO: what about if s is a tensor and a no tensor expression?
-        my_diffs = [a._eval_partial_derivative(s)
-                    if isinstance(a, TensExpr) else a.diff(s)
-                    for a in self.args]
+        my_diffs = []
+        for a in self.args:
+            if isinstance(a, TensExpr):
+                my_diffs.append(a._eval_partial_derivative(s))
+            else:
+                # do not call diff if s is no symbol
+                if isinstance(s, Symbol):
+                    my_diffs.append(a.diff(s))
```

Comment:
`a._eval_derivative(s)`... `a.diff(s)` is quite expensive.

Reply:
Good to know. That's my missing knowledge of the underlying API :-) Thanks for fixing.